src/data/polynomial/unit_trinomial.lean

```diff
@@ -0,0 +1,315 @@
+/-
+Copyright (c) 2022 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import analysis.complex.polynomial
+import data.polynomial.mirror
+import ring_theory.roots_of_unity
+
+/-!
+# Unit Trinomials
+
+This file defines irreducible trinomials and proves an irreducibility criterion.
+
+## Main definitions
+
+- `polynomial.is_unit_trinomial`
+
+## Main results
+
+- `polynomial.irreducible_of_coprime`: An irreducibility criterion for unit trinomials.
+
+TODO: Irreducibility of x^n-x-1
+-/
+
+namespace polynomial
+open_locale polynomial
+
+open finset
+
+section semiring
+
+variables {R : Type*} [semiring R] {k m n : ℕ} {u v w : R}
+
+lemma trinomial_leading_coeff' (hkm : k < m) (hmn : m < n) :
+  (C u * X ^ k + C v * X ^ m + C w * X ^ n).coeff n = w :=
```

Comment:
What do you think about defining `trinomial hkm hmn u v w := C u * X ^ k + C v * X ^ m + C w * X ^ n`?